webapi/src/main/scala/org/knora/webapi/messages/v2/responder/ontologymessages/KnoraApiV2Simple.scala

```diff
@@ -0,0 +1,1114 @@
+/*
+ * Copyright © 2015 Lukas Rosenthaler, Benjamin Geer, Ivan Subotic,
+ * Tobias Schweizer, André Kilchenmann, and Sepideh Alassi.
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.knora.webapi.messages.v2.responder.ontologymessages
+
+import org.knora.webapi._
+
+/**
+  * Represents the `knora-api` ontology, version 2, in the [[ApiV2Simple]] schema.
+  */
+object KnoraApiV2Simple {
+
+    val Resource: ClassEntityInfoV2 = makeClass(
+        classIri = OntologyConstants.KnoraApiV2Simple.Resource,
+        cardinalities = Map(
+            OntologyConstants.SchemaOrg.Name -> Cardinality.MustHaveOne,
+            OntologyConstants.KnoraApiV2Simple.HasStandoffLinkTo -> Cardinality.MayHaveMany,
+            OntologyConstants.KnoraApiV2Simple.CreationDate -> Cardinality.MustHaveOne,
+            OntologyConstants.KnoraApiV2Simple.LastModificationDate -> Cardinality.MayHaveOne
+        )
+    )
+
+    val HasStandoffLinkTo: PropertyEntityInfoV2 = makeProperty(
+        propertyIri = OntologyConstants.KnoraApiV2Simple.HasStandoffLinkTo,
+        propertyType = OntologyConstants.Owl.ObjectProperty,
+        subPropertyOf = Set(OntologyConstants.KnoraApiV2Simple.HasLinkTo),
+        predicates = Seq(
+            makePredicate(
+                predicateIri = OntologyConstants.Rdfs.Label,
+                objectsWithLang = Map(
+                    LanguageCodes.DE -> "hat Standofflink zu",
```

Comment:
Do you see a practical possibility not to define rdfs:label and rdf:comment on entities (classes and properties) redundantly in `knora-base.ttl` and in  `KnoraApiV2Simple.scala` as well as in  `KnoraApiV2WithValueObjects.scala`?

Reply:
Since the ontology cache is built on startup and all of knora-base is loaded, couldn't you just get this information form the cache (given the IRI)?

Reply:
I don't think so, because the ontology cache contains `knora-base`, which doesn't have the same structure as the `knora-api` ontologies. There are properties and classes in `knora-api` that have no equivalent in `knora-base`. Also, some of the "same" classes have a different structure (e.g. `DateValue`). So it makes sense that the labels and comments could be different. For example, the comment for `knora-base:DateValue` should say that it stores Julian Day Numbers, but not the comment for `knora-api:DateValue`.

Also, the objects in `KnoraApiV2Simple` and `KnoraApiV2WithValueObjects` are constructed when the JVM starts, before the ontology responder starts.

Reply:
I get the point when things are actually different, but wouldn't it make sense to share information if they are the same?

Also I do not understand why `knora-api:Region` has to be hard-coded. Like any other resource class defined in a project-ontology,  it is a subclass of `knora-base:Resource` and we already have the mechanisms to convert that to `knora-api`.

I do understand why this is necessary for value classes though.

Reply:
for instance, the current implementation knora-api:Region in complex JSONLD is incomplete (label and comment are missing):

```
{
@id: "knora-api:Region",
@type: "owl:Class",
knora-api:belongsToOntology: "http://api.knora.org/ontology/knora-api/v2",
knora-api:canBeInstantiated: true,
rdfs:subClassOf: [
"http://api.knora.org/ontology/knora-api/v2#Resource",
{
@type: "owl:Restriction",
owl:maxCardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#lastModificationDate"
},
{
@type: "owl:Restriction",
owl:cardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#hasColor"
},
{
@type: "owl:Restriction",
owl:minCardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#hasGeometry"
},
{
@type: "owl:Restriction",
owl:minCardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#hasComment"
},
{
@type: "owl:Restriction",
owl:cardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#creationDate"
},
{
@type: "owl:Restriction",
owl:cardinality: 1,
owl:onProperty: "http://schema.org/name"
},
{
@type: "owl:Restriction",
owl:cardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#hasPermissions"
},
{
@type: "owl:Restriction",
owl:minCardinality: 0,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#hasStandoffLinkTo"
},
{
@type: "owl:Restriction",
owl:cardinality: 1,
owl:onProperty: "http://api.knora.org/ontology/knora-api/v2#isRegionOf"
}
]
},
```

In general, our documentation is already poor (labels and comments are sometimes not available in all supported languages), and this won't improve if we have to do it in three different places.

Reply:
This would mean implementing some way of deciding whether to use a hard-coded API entity definition or to convert a `knora-base` definition. How would that decision be made?

Also, `knora-base:Region` has predicates that `knora-api:Region` doesn't have, and vice versa.